ivadomed/testing.py

```diff
@@ -236,11 +224,17 @@ def run_inference(test_loader, model, model_params, testing_params, ofolder, cud
                     preds_npy_list.append(output_nii.get_fdata().transpose(3, 0, 1, 2))
                     gt_lst = []
                     for gt in metadata[0]['gt_filenames']:
-                        gt_lst.append(nib.load(gt).get_fdata())
+                        # For multi-label, if all labels are not in every image
+                        if gt is not None:
+                            gt_lst.append(nib.load(gt).get_fdata())
+                        else:
+                            gt_lst.append(np.zeros(gt_lst[0].shape))
+
                     gt_npy_list.append(np.array(gt_lst))
                     # Save merged labels with color
-                    if volume.shape[0] > 1:
-                        imed_utils.save_color_labels(pred_undo,
+
+                    if pred_undo.shape[0] > 1:
+                        imed_utils.save_color_labels(threshold_predictions(pred_undo, thr=0.5),
```

Comment:
Why do you need to thr before calling the function while we have the parameter "binarize_predictions" as well?